File: Python/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        n = len(strs)
        mp=defaultdict(list)
        s=""

        for str in strs:
           sorted_str = s.join(sorted(str))
           mp[sorted_str].append(str)
        
        return sorted(mp.values())

File: C++/49. Group Anagrams.cpp
class Solution {
public:
       vector<vector<string>> groupAnagrams(vector<string>& strs) {
           int n = strs.size();
           unordered_map<string,vector<string>> mp;
           vector<vector<string>> sol;
           for(int i = 0;i<n;i++){
               string str = strs[i];
               sort(str.begin(),str.end());
               mp[str].push_back(strs[i]);
            }
           for(auto i:mp) sol.push_back(i.second);
           return sol;
    }
};
